drv/lpc55-swd/build.rs

```diff
@@ -0,0 +1,89 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::Result;
+use build_lpc55pins::PinConfig;
+use quote::ToTokens;
+use serde::Deserialize;
+use std::io::Write;
+
+#[derive(Deserialize)]
+struct TaskConfig {
+    in_cfg: Vec<PinConfig>,
+    out_cfg: Vec<PinConfig>,
+    pins: Vec<PinConfig>,
+    spi_num: usize,
+}
+
+fn generate_swd_functions(config: &TaskConfig) -> Result<()> {
```

Comment:
I've done this in a couple cases, but one warning: `quote!` discards comments. So your io_out/io_in comments there would be visible in the code generator, but not the generated code. This may or may not concern you.

Reply:
Oh also the display impl for `quote` that you used there prints all generated code on one line. I've got the `lib/call_rustfmt` crate if you want to fix that.